server/shared/src/main/scala/org/http4s/server/middleware/RequestId.scala

```diff
@@ -44,53 +45,104 @@ object RequestId {
 
   val requestIdAttrKey: Key[String] = Key.newKey[SyncIO, String].unsafeRunSync()
 
-  def apply[G[_], F[_]](http: Http[G, F])(implicit G: Sync[G]): Http[G, F] =
+  def apply[G[_]: Sync, F[_]](http: Http[G, F]): Http[G, F] =
     apply(requestIdHeader)(http)
 
   @deprecated("Preserved for bincompat", "0.23.12")
   def apply[G[_], F[_]](headerName: CIString, http: Http[G, F], G: Sync[G]): Http[G, F] =
     apply(headerName)(http)(G, UUIDGen.fromSync(G))
 
+  @deprecated("Preserved for bincompat", "0.23.30")
   def apply[G[_], F[_]](
+      headerName: CIString,
+      http: Http[G, F],
+      G: Sync[G],
+      gen: UUIDGen[G],
+  ): Http[G, F] =
+    apply(headerName)(http)(G, gen)
+
+  def apply[G[_]: Monad: UUIDGen, F[_]](
       headerName: CIString
-  )(http: Http[G, F])(implicit G: Sync[G], gen: UUIDGen[G]): Http[G, F] =
-    Kleisli[G, Request[F], Response[F]] { req =>
-      for {
-        header <- req.headers.get(headerName).map(_.head) match {
-          case None => UUIDGen.randomString[G].map(Header.Raw(headerName, _))
-          case Some(header) => G.pure(header)
-        }
-        reqId = header.value
-        response <- http(req.withAttribute(requestIdAttrKey, reqId).putHeaders(header))
-      } yield response.withAttribute(requestIdAttrKey, reqId).putHeaders(header)
-    }
+  )(http: Http[G, F]): Http[G, F] =
+    apply(headerName, UUIDGen.randomUUID[G])(http)
 
+  @deprecated("Preserved for bincompat", "0.23.30")
```

Comment:
v0.23.30 was already released.

```suggestion
  @deprecated("Preserved for bincompat", "0.23.31")
```

Reply:
Updated.